### PR TITLE
Convert TUITextRenderer autoscrolling rects to the scroll view's coordinate system

### PR DESCRIPTION
**Pull request by jspahrsummers**
CC @galaxas0

### CODE REVIEW

**Comment by avaidyam** [comment]
Oh thanks, do I merge this? The github email tells me something about that.
EDIT: Just kidding, I can't merge this. LOL.

File: lib/UIKit/TUITextRenderer+KeyBindings.m [hunk 1]
```diff
@@ -80,22 +80,23 @@ - (TUITextEditor *)_textEditor
 }
 
 - (void)_scrollToIndex:(CFIndex)index {
-	id <TUIBridgedScrollView> scrollView = self.view.ancestorScrollView;
-	if(scrollView) {
-		
+	id<TUIBridgedScrollView> scrollView = self.view.ancestorScrollView;
+	if (scrollView) {
```
Comment (avaidyam) [inline lib/UIKit/TUITextRenderer+KeyBindings.m:84]: Eek, I generally don't like the `id` and the `<protocol>` being right next to each other, but the `if` and the `(statement)` together.

File: lib/UIKit/TUITextRenderer+KeyBindings.m [hunk 2]
```diff
@@ -80,22 +80,23 @@ - (TUITextEditor *)_textEditor
...
-		// If it exists, then scroll the the beginning of the rects.
-		if(nRects == 1)
-			[scrollView scrollToIncludeRect:rects[0]];
-		
-		// Otherwise, scroll to the middle of the text rects.
-		else if(nRects == 0)
-			[scrollView scrollToIncludeRect:CGRectMake(0, self.view.frame.size.height / 2, 0, 1)];
+		if (nRects == 1) {
+			// If it exists, then scroll the the beginning of the rects.
+			CGRect rect = [scrollView convertFromWindowRect:[self.view convertToWindowRect:rects[0]]];
```
Comment (avaidyam) [inline lib/UIKit/TUITextRenderer+KeyBindings.m:93]: Do the `ancestorScrollView` methods call for window rects?

File: lib/UIKit/TUITextRenderer+KeyBindings.m (hunk 1 above)
Comment (alanjrogers) [inline lib/UIKit/TUITextRenderer+KeyBindings.m:84]: This style matches our [coding guidelines](https://github.com/github/objective-c-conventions#control-structures).  If you don't agree with the style feel free to start a discussion on that repository.  But be prepared to argue the case :smile:

File: lib/UIKit/TUITextRenderer+KeyBindings.m [hunk 3]
```diff
@@ -80,22 +80,23 @@ - (TUITextEditor *)_textEditor
...
 		
-		// If it exists, then scroll the the beginning of the rects.
-		if(nRects == 1)
-			[scrollView scrollToIncludeRect:rects[0]];
-		
-		// Otherwise, scroll to the middle of the text rects.
-		else if(nRects == 0)
-			[scrollView scrollToIncludeRect:CGRectMake(0, self.view.frame.size.height / 2, 0, 1)];
+		if (nRects == 1) {
+			// If it exists, then scroll the the beginning of the rects.
```
Comment (alanjrogers) [inline lib/UIKit/TUITextRenderer+KeyBindings.m]: *to 

:trollface:

**Comment by alanjrogers** [comment]
:postbox:

Just the one note from me @jspahrsummers

File: lib/UIKit/TUITextRenderer+KeyBindings.m (hunk 2 above)
Comment (jspahrsummers) [inline lib/UIKit/TUITextRenderer+KeyBindings.m:93]: There's no knowing how high up in the view hierarchy the scroll view may be. You have to convert to its coordinate system somehow, and this is the only conversion method guaranteed to be implemented by `<TUIBridgedScrollView>`.

**Comment by jspahrsummers** [comment]
@alanjrogers :zap:

**Comment by dannygreg** [comment]
This looks :fire: to me. Finna merge.

**Pull request merged**